zjit/src/hir_type/gen_hir_type.rb

```diff
@@ -74,19 +74,21 @@ def base_type name
 base_type "Range"
 base_type "Set"
 base_type "Regexp"
-base_type "Class"
+_, module_exact = base_type "Module"
+module_exact.subtype "Class"
```

Comment:
Since `Class` is final we can remove `ClassExact` and just use that (consistent with `Fixnum`, etc)

Reply:
Also we should make `Class` subclass `Module`, not `ModuleExact`

Reply:
Updated